src/map/skill.cpp

```diff
@@ -3414,48 +3414,49 @@ static void skill_do_copy(struct block_list* src,struct block_list *bl, uint16 s
 		if (!(idx = skill_get_index(skill_id)))
 			return;
 
-		switch (skill_isCopyable(tsd,skill_id)) {
-			case 1: //Copied by Plagiarism
-				{
-					if (tsd->cloneskill_idx > 0 && tsd->status.skill[tsd->cloneskill_idx].flag == SKILL_FLAG_PLAGIARIZED) {
-						clif_deleteskill(*tsd,tsd->status.skill[tsd->cloneskill_idx].id, true);
-						tsd->status.skill[tsd->cloneskill_idx].id = 0;
-						tsd->status.skill[tsd->cloneskill_idx].lv = 0;
-						tsd->status.skill[tsd->cloneskill_idx].flag = SKILL_FLAG_PERMANENT;
+		switch ( skill_isCopyable(tsd, skill_id) ) {
+			case 1: { //Copied by Plagiarism
+					//Delete reproduced skill when the plagiarized skill id is the same
+					if ( tsd->reproduceskill_idx && tsd->status.skill[tsd->reproduceskill_idx].flag == SKILL_FLAG_PLAGIARIZED && tsd->status.skill[tsd->reproduceskill_idx].id == skill_id ) {
+						pc_skill_plagiarism_reset(*tsd, 2);
 					}
 
-					lv = min(skill_lv,pc_checkskill(tsd,RG_PLAGIARISM)); //Copied level never be > player's RG_PLAGIARISM level
+					pc_skill_plagiarism_reset(*tsd, 1);
+
+					//Cap level to RG_PLAGIARISM level
+					lv = min(skill_lv, pc_checkskill(tsd, RG_PLAGIARISM));
 
 					tsd->cloneskill_idx = idx;
 					pc_setglobalreg(tsd, add_str(SKILL_VAR_PLAGIARISM), skill_id);
 					pc_setglobalreg(tsd, add_str(SKILL_VAR_PLAGIARISM_LV), lv);
-				}
+				} 
 				break;
-			case 2: //Copied by Reproduce
-				{
+
+			case 2: { //Copied by Reproduce
+					//Delete plagiarized skill when the reproduced skill id is the same
+					if ( tsd->cloneskill_idx > 0 && tsd->status.skill[tsd->cloneskill_idx].flag == SKILL_FLAG_PLAGIARIZED && tsd->status.skill[tsd->cloneskill_idx].id == skill_id ) {
+						pc_skill_plagiarism_reset(*tsd, 1);
+					}
+
 					status_change *tsc = status_get_sc(bl);
-					//Already did SC check
-					//Skill level copied depends on Reproduce skill that used
```

Comment:
Do we really want to remove this comment? (Only talking about "Skill level copied depends on Reproduce skill that used")